### ISSUE TITLE
multilabel_accuracy yields wrong results compared to scikit-learn

### ISSUE THREAD

**Issue by xor-xor** [Lightning-AI/torchmetrics#1853]
## 🐛 Bug

Hi, I've found a significant difference in results between TM's `multilabel_accuracy` and scikit-learn's `accuracy_score`, and given that [you guys aim to match the behavior of the latter](https://github.com/Lightning-AI/torchmetrics/issues/298#issuecomment-874568132), this probably should be considered as a bug.

From my brief examination, it seems that the root of the problem lies in this statement from [scikit-learn's docs](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html#sklearn.metrics.accuracy_score) for `accuracy_score`:

> In multilabel classification, this function computes subset accuracy: the set of labels predicted for a sample must _exactly_ match the corresponding set of labels in y_true.

...which doesn't seem to be true for TM's version.

### To Reproduce

```python
import numpy as np
import torch
from sklearn.metrics import accuracy_score
from torchmetrics.functional.classification import multilabel_accuracy

rng = np.random.default_rng(42)

num_labels = 200
num_examples = 20000
preds = rng.integers(0, 2, (num_examples, num_labels))
targets = rng.integers(0, 2, (num_examples, num_labels))

multilabel_accuracy(torch.from_numpy(preds), torch.from_numpy(targets), num_labels).item()
# >>> 0.5000045299530029

accuracy_score(targets, preds)
# >>> 0.0

((targets == preds).sum(axis=1) == num_labels).sum() / num_examples
# >>> 0.0


num_labels = 2
num_examples = 20000
preds = rng.integers(0, 2, (num_examples, num_labels))
targets = rng.integers(0, 2, (num_examples, num_labels))

multilabel_accuracy(torch.from_numpy(preds), torch.from_numpy(targets), num_labels).item()
# >>> 0.4971500039100647

accuracy_score(targets, preds)
# >>> 0.2475

((targets == preds).sum(axis=1) == num_labels).sum() / num_examples
# >>> 0.2475
```

### Expected behavior

To have results consistent with scikit-learn.

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): `0.11.4` (installed via `pip`)
- Python & PyTorch Version (e.g., 1.0): `3.10.6` & `1.31.1`
- Any other relevant information such as OS (e.g., Linux): macOS `13.3.1`

### Additional context

None.

**Comment by SkafteNicki** [comment]
Hi @xor-xor, thanks for raising this issue.
You are completely right that `multilabel_accuracy` does not compute subset accuracy. It computes, as the name suggest just accuracy. If you want the subset accuracy it can be done using the `exact_match` metric (exact match is another commonly used naming instead of subset accuracy):
https://torchmetrics.readthedocs.io/en/stable/classification/exact_match.html
```python
# in your example try adding 
from torchmetrics.functional.classification import multilabel_exact_match
print(multilabel_exact_match(torch.from_numpy(preds), torch.from_numpy(targets), num_labels).item())
```
which give the same value.

I can see how my old comment is coming back to haunt me now, but I think the context is important. The core idea here is to split subset accuracy (or exact match) into its own metric, because we seen in the past that users are both interested in calculating the normal accuracy and subset accuracy for multilabel problems.

**Comment by xor-xor** [comment]
@SkafteNicki thanks for your prompt and informative answer! It seems that my ignorance is at fault here, because I simply wasn't aware that "subset accuracy" is also known as "exact match" metric (otherwise it would definitely catch my eye while browsing the list of TM's metrics)  - sorry about that! Actually, when I was composing this bug report, I had this lingering thought: "this can't be a bug, because it would be too big given how obvious the difference is - but on the other hand, where does that difference comes from?" ;) And yes, I agree that splitting those two metrics totally makes sense!
